.github/workflows/CI.yml

```
@@ -96,6 +96,50 @@ jobs:
           fi
         shell: bash
 
+  snapshots:
+    name: Snapshots
+    runs-on: ubuntu-latest
+    if: ${{ !github.event.pull_request.draft }}
+    env:
+      AMARU_PEER_ADDRESS: preprod-node.play.dev.cardano.org:3001
+    steps:
+      - uses: actions/checkout@v4
+      - uses: Swatinem/rust-cache@v2
+      - name: Cache ledger.db
+        id: cache
+        uses: actions/cache@v3
+        with:
+          path: ~/ledger.db
+          key: ${{ runner.OS }}-ledger-cache-${{ hashFiles('.ledger.db/**') }}
+          restore-keys: |
+            ${{ runner.OS }}-ledger-cache-
```

```suggestion
          key: ${{ runner.OS }}-ledger-cache-v1
          restore-keys: |
            ${{ runner.OS }}-ledger-cache-
```

We most likely don't want to upload the new ledger.db file to the cache since part of the test is going to be about re-generating those snapshots. Instead, we simply want to download the _initial snapshots_, and run Amaru from there.